CI: getting started with GitHub Actions

pzhlkj6612 opened the pull request:
Hello!

<br>

In [README.md](https://github.com/antonmedv/countdown/blob/d031d02df559feeb55f5449701df4b0d27950ace/README.md), we have:

> ...
> Or download prebuilt binary from [releases](https://github.com/antonmedv/countdown/releases).

However, only the "[v1.0.0](https://github.com/antonmedv/countdown/releases/tag/v1.0.0)" release contains pre-built binaries, so I've made a workflow to make snapshot releases.

Please note that not all `GOOS/GOARCH` pairs are supported by GitHub Actions, so it may not be possible to build for every platform supported by Go. The `android/arm64` build works on my phone.

pzhlkj6612 committed:
ci: build and release snapshots with GitHub Actions

pzhlkj6612 committed:
chore: Go to version 1.20

antonmedv commented:
As I release not very often I decided not to go with CI. 

But countdown got much more popular than anticipated. Probably a good time for a CI

Let’s implement something similar like in my other projects: https://github.com/antonmedv/llama/blob/master/.github/workflows/release.mjs

pzhlkj6612 commented:
Oh, it seems that I've just made a few things about releasing a new countdown... I need to learn the workflow in `antonmedv/llama` first.

pzhlkj6612 committed:
merge master

pzhlkj6612 committed:
CI builds assets with zx; import brew workflow; Go 1.23; update docs

- No Android, too difficult.
- "install.sh" does not make so much sense on Windows.

pzhlkj6612 commented:
Hey!

> Let’s implement something similar like in my other projects: https://github.com/antonmedv/llama/blob/master/.github/workflows/release.mjs

Done. I've been using https://github.com/antonmedv/walk/blob/c78d2116a868f5471171c0f5a8d36243e44298eb/.github/scripts/build.mjs as the base.

For now:

- Building on Android is difficult. We need the help from https://github.com/go-cross/cgo-actions if we'd like to do so.
- "install.sh" does not make so much sense on Windows. Shall we keep it?

.

antonmedv commented on .github/workflows/build_and_release.yml:
```diff
@@ -0,0 +1,33 @@
+        run: |
+          set -euo pipefail
+
+          npm install zx
```
Npm install not needed for npx

pzhlkj6612 replied:
You're right! 5f783099d6820279757e73180e35b53d858da1cf.

pzhlkj6612 committed:
Npm install not needed for npx

Co-authored-by: Anton Medvedev <anton@medv.io>
